### PR TITLE
Introduce JSX and Composition API

### PR DESCRIPTION
**Pull request by richardgoater**
Demonstrating integration of these two features. I opted not to replicate slots/scoped slots in the Element UI table, instead re-implementing as a simple HTML table.

@LachlanStuart let's check if the config changes don't cause you any issues in Webstorm.

### CODE REVIEW

**Comment by richardgoater** [comment]
... it's breaking CI apparently

**Comment by richardgoater** [comment]
It's now failing on a snapshot, but I would say the new one looks more correct - it's actually rendering data. Would appreciate a second opinion.

File: metaspace/webapp/src/modules/Annotations/AnnotationTable.spec.ts [hunk 1]
```diff
@@ -21,9 +21,6 @@ const blobToText = (blob: Blob) => new Promise<string>((resolve, reject) => {
   reader.readAsText(blob)
 })
 
-// HACK: Prevent AnnotationTable from waiting before loading data
-delete (AnnotationTable as any).apollo.annotations.debounce
```
Comment (LachlanStuart) [inline metaspace/webapp/src/modules/Annotations/AnnotationTable.spec.ts]: Need to re-add this to fix the snapshot tests:
```
// HACK: Prevent AnnotationTable from waiting before loading data
delete (AnnotationTable as any).options.apollo.annotations.debounce
```

**Comment by LachlanStuart** [comment]
Also, don't worry about the "codecov" test failing - we mostly use it for tracking the coverage trend and as a reminder when we should consider adding more tests. However, quite often it's a false alarm, because it sometimes compares against previous commits in the same PR instead of comparing against the coverage in the master branch.

File: metaspace/webapp/src/modules/UserProfile/Table.css [hunk 2]
```diff
@@ -0,0 +1,38 @@
+.sm-table {
+  width: 100%;
+  color: #909399;
+  font-size: 14px;
+  border-spacing: 0;
+}
```
[the remark this answers is not in the document]
Comment (richardgoater) [reply metaspace/webapp/src/modules/UserProfile/Table.css]: Thanks dude, this gives me a much better appreciation of the edge cases.

Hopefully this is looking better:

Screen Shot 2020-02-14 at 14 00 13-fullpage

**Comment by richardgoater** [comment]
@intsco there seem to be some commits from you in this PR, not sure if this is ok?

**Comment by LachlanStuart** [comment]
@richardgoater It looks like you based this branch off of the branch for PR #520, which has since been merged.  Ideally you'd rebase this branch off of those commits on to master. It's optional this time though - the files changed by those commits exactly match the files in master, so GitHub isn't complaining because it can merge without conflicts.

**Pull request merged**